### DISCUSSION TITLE
FR: Short name for long page titles in topnav

### DISCUSSION THREAD

**Discussion by metbril** [Lednerb/bilberry-hugo-theme#437] [Ideas]
I use long titles for my pages. And I have multiple. Those all appear in the topnav, which get crowded like this. For example, I have a page '/now' with the title 'What I am doing now' and a '/contact' page titled 'How to contact me'.

Other themes allow me to have a 'name' parameter with the short name. This short name is used in the menus, where the title can be used to show at the top of the page.

In any page, this could be added in the front matter:

```markdown
---
title: What I am doing now
slug: now
menu:
  main:
    name: Now
---
Etc
```

The topnav partial could be adjusted accordingly, to use `{{ .Name }}` instead of `{{ .Title }}`.
For backwards compatibility, Title could still be used as a fallback value if Name is not specified.

**Comment by metbril** [comment]
After reviewing the code, I noticed that the theme does not use the standard Hugo way for menus. The fix would be to add a frontmatter parameter `name:` to the page like this:

```
---
title: How to contact me
name: Contact
---
Etc.
```

and then replace one line in the `topnav.html` partial from

```html
                    <li><a href="{{ .Permalink }}">{{ .Title }}</a></li>
```

to

```html
                    <li><a href="{{ .Permalink }}">{{ .Params.Name | default .Title }}</a></li>
```

**Comment by igor-baiborodine** [comment]
@metbril, I'm moving it to Discussions/Ideas since it should be considered as an enhancement.
